WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java

```diff
@@ -173,32 +172,8 @@ public static ReaderPost fromJson(JSONObject json) {
         // xpost info
         assignXpostIdsFromJson(post, json.optJSONArray("metadata"));
 
-        // if there's no featured image, check if featured media has been set - this is sometimes
-        // a YouTube or Vimeo video, in which case store it as the featured video so we can treat
-        // it as a video
-        if (!post.hasFeaturedImage()) {
-            JSONObject jsonMedia = json.optJSONObject("featured_media");
-            if (jsonMedia != null && jsonMedia.length() > 0) {
-                String mediaUrl = JSONUtils.getString(jsonMedia, "uri");
-                if (!TextUtils.isEmpty(mediaUrl)) {
-                    String type = JSONUtils.getString(jsonMedia, "type");
-                    boolean isVideo = (type != null && type.equals("video"));
-                    if (isVideo) {
-                        post.featuredVideo = mediaUrl;
-                    } else {
-                        post.featuredImage = mediaUrl;
-                    }
-                }
-            }
-        }
-        // if the post still doesn't have a featured image but we have attachment data, check whether
-        // we can find a suitable featured image from the attachments
-        if (!post.hasFeaturedImage() && post.hasAttachments()) {
-            post.featuredImage = new ImageSizeMap(post.getText(), post.attachmentsJson)
-                    .getLargestImageUrl(ReaderConstants.MIN_FEATURED_IMAGE_WIDTH);
-        }
-        // if we *still* don't have a featured image but the text contains an IMG tag, check whether
-        // we can find a suitable image from the text
+        // if the post doesn't have a featured image but it contains an IMG tag, check whether
+        // we can find a suitable image from the content
         if (!post.hasFeaturedImage() && post.hasImages()) {
             post.featuredImage = new ReaderImageScanner(post.text, post.isPrivate)
```

Comment:
I totally agree having the right context gives words the right meaning. In this case given the most prominent concept in the WP world (that is, the shared understanding we all have) for the "featured image" tag/label/name refers to the concept of an image _set by the user_, 2 levels of indirection are needed in order to arrive at the meaning of the name `featuredImage` within this particular sub-context. My take is that here or anywhere else, it shouldn't be necessary to have any special experience working on the feature (like you guys obviously have) to be able to tell the difference - code should be easy to read and this particular case has inherent ambiguity in it.

But, all that said, it doesn't make practical sense to extend this discussion any longer - not without some beers at least :)

So I'll just repeat what I said before

> Having that in mind, LGTM

☝️ :shipit: 🍻